python/src/open_discourse/steps/speech_content/extract.py

```diff
@@ -19,18 +19,67 @@ def main(task):
     print("Starting..")
 
     for folder_path in sorted(RAW_TXT.iterdir()):
-        process_period(folder_path)
+
+        # Get regex helpers based on folder name/term
+        open_brackets, close_brackets, prefix = get_bracket_and_prefix_from_term_number(folder_path)
+        if open_brackets is None:
+            continue  # skip if not a valid folder
+
+        president_pattern = get_president_pattern()
+
+        faction_speaker_pattern = get_faction_speaker_pattern(
+            term_number=int(folder_path.stem[-2:]),
+            folder_path=folder_path,
+            open_brackets=open_brackets,
+            close_brackets=close_brackets,
+            prefix=prefix
+        )
+
+        minister_pattern = get_minister_pattern(open_brackets, close_brackets, prefix)
+
+        process_period(folder_path, president_pattern, faction_speaker_pattern, minister_pattern)
 
     return True
 
+def get_bracket_and_prefix_from_term_number(folder_path: Path):
```

Comment:
if you start adding type hints you should also add the return type of the function, as it is needed for a complete type hint:
`get_bracket_and_prefix_from_term_number(folder_path: Path) -> tuple[str|None]`
so probably we should return empty string "" in case of an error and not None so you can rely on always finding string in the tuple